ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+    if (hMiraMonLayer->ReadOrWrite == MM_WRITTING_MODE)
+    {
+        if (MMInitFlush(&pZSection->FlushZL, pF3d,
+                        (hMiraMonLayer->nMemoryRatio != 1)
+                            ? (GUInt64)(hMiraMonLayer->nMemoryRatio * MM_25MB)
+                            : MM_25MB,
+                        &pZSection->pZL, 0, sizeof(double)))
+            return 1;
+    }
+    return 0;
+}
+
+/* -------------------------------------------------------------------- */
+/*      Layer Functions: Extensions                                     */
+/* -------------------------------------------------------------------- */
+
+/* Find the last ocurrence of pszFinalPart in pszName
+    and changes it by pszNewPart.
+ 
+    Examples of desired behaviour
+    AA.pnt -> AAT.rel
+    AA.nod -> N.~idx
+    AA.nod -> N.dbf
+    AA.nod -> N.rel
+*/
+
+static int MMChangeFinalPartOfTheName(char *pszName, size_t nMaxSizeOfName,
+                                      const char *pszFinalPart,
+                                      const char *pszNewPart)
+{
+    char *pAux, *pszWhereToFind, *pszLastFound = nullptr;
+    ;
+
+    if (!pszName || !pszFinalPart || !pszNewPart)
+        return 0;
+    if (MMIsEmptyString(pszName) || MMIsEmptyString(pszFinalPart) ||
+        MMIsEmptyString(pszNewPart))
+        return 0;
+
+    if (strlen(pszName) - strlen(pszFinalPart) + strlen(pszNewPart) >=
+        nMaxSizeOfName)
+        return 1;  // It's not possible to change the final part
+
+    pszWhereToFind = pszName;
+    while (nullptr != (pAux = strstr(pszWhereToFind, pszFinalPart)))
+    {
+        pszLastFound = pAux;
+        pszWhereToFind = pAux + strlen(pAux);
+    }
+
+    if (!pszLastFound)
+        return 1;  // Not found not changed
+
+    memcpy(pszLastFound, pszNewPart, strlen(pszNewPart));
+
+    return 0;
+}
+
+/* -------------------------------------------------------------------- */
+/*      Layer Functions: initializing MiraMon layers                    */
+/* -------------------------------------------------------------------- */
+static int MMInitPointLayer(struct MiraMonVectLayerInfo *hMiraMonLayer)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    hMiraMonLayer->bIsPoint = 1;
+
+    if (hMiraMonLayer->ReadOrWrite == MM_WRITTING_MODE)
+    {
+        // Geometrical part
+        // Init header structure
+        hMiraMonLayer->TopHeader.nElemCount = 0;
+        MMInitBoundingBox(&hMiraMonLayer->TopHeader.hBB);
+
+        hMiraMonLayer->TopHeader.bIs3d = 1;  // Read description of bRealIs3d
+        hMiraMonLayer->TopHeader.aFileType[0] = 'P';
+        hMiraMonLayer->TopHeader.aFileType[1] = 'N';
+        hMiraMonLayer->TopHeader.aFileType[2] = 'T';
+
+        // Opening the binary file where sections TH, TL[...] and ZH-ZD[...]-ZL[...]
+        // are going to be written.
+
+        strcpy(hMiraMonLayer->MMPoint.pszLayerName,
+               hMiraMonLayer->pszSrcLayerName);
+        strcat(hMiraMonLayer->MMPoint.pszLayerName, ".pnt");
```

Comment:
isn't there any risk of this to overrun MMPoint.pszLayerName ? If so, then maybe
```suggestion
        snprintf(hMiraMonLayer->MMPoint.pszLayerName, sizeof((hMiraMonLayer->MMPoint.pszLayerName), "%s.pnt", hMiraMonLayer->pszSrcLayerName);
```
Same remark for similar patterns below

Reply:
Yes, changed at all places.